src/transaction.js

```diff
@@ -731,21 +738,30 @@ const transaction = {
     try {
       data = transaction.prepareData(data, pin, options);
     } catch (e) {
-      if (e instanceof AddressError ||
-          e instanceof OutputValueError ||
-          e instanceof ConstantNotSet ||
-          e instanceof CreateTokenTxInvalid ||
-          e instanceof MaximumNumberOutputsError ||
-          e instanceof MaximumNumberInputsError) {
-        return Promise.reject(e.message);
-      } else {
-        // Unhandled error
-        throw e;
-      }
+      const message = helpers.handlePrepareDataError(e);
+      return Promise.reject(e.message);
```

Comment:
Should it be just `message`?

Reply:
Not in this method. This is the old send transaction method and it always returns a promise.

Reply:
Why do we need `const message = helpers.handlePrepareDataError(e); `? We don't use it.

Reply:
It's like @msbrogli said, it should be `message`. Fixed it, thanks.